Change the module commands that load pe_archive to be pe_archive/append

ndkeen opened the pull request:
Change the module commands that load pe_archive to be pe_archive/append-path

NERSC made a change that broke our environment and a fix was suggested to use "pe_archive/append-path" instead of "pe_archive".
This allows for the use of older versions of modules.

This is primarily for edison so that we can continue using cray-mpich/7.6.0.
Also changed the instances for cori so that we can use gcc/6.3.0 -- obviously only affects cori builds using the GNU compiler.

Fixes https://github.com/E3SM-Project/E3SM/issues/2829

[bfb]

ndkeen commented:
merged to maint-1.0

salilmahajan commented:
Hi @ndkeen, I just tried to build on edison and cori with the maint1.0 branch and I am getting the same/similar errors again:

run_e3sm: -------- Starting case.setup --------

run_e3sm: /scratch1/scratchdirs/salil/E3SM_simulations/default.default.edison_ne4_test_04012019.ne4np4_oQU240/case_scripts/case.setup
ERROR: module command /opt/modules/default/bin/modulecmd python load pe_archive/append-path cray-mpich/7.6.0 PrgEnv-intel/6.0.4 failed with message:
utility.c(2263):ERROR:50: Cannot open file '/usr/common/software/modulefiles/pe_archive/.version' for 'reading'
locate_module.c(1241):ERROR:106: Magic cookie '#%Module' missing in '/usr/common/software/modulefiles/pe_archive/.version'

Any suggestions? Thanks!

ndkeen commented:
Thanks for reporting -- I just tried to load the module from command line now and it does appear to be broken. Maybe not in the same way as before, but still not working right. I sent an email to NERSC. I can maybe guess that there is general software maintenance happening which is maybe leaving these older versions in erratic state.

```
edison05% module load pe_archive/append-path
utility.c(2263):ERROR:50: Cannot open file '/usr/common/software/modulefiles/pe_archive/.version' for 'reading'
locate_module.c(1241):ERROR:106: Magic cookie '#%Module' missing in '/usr/common/software/modulefiles/pe_archive/.version'
```
Note that with any E3SM repo, you can test out the environment with the below command. If this doesn't work, the build will likely not work. It also helps to verify that the issue is within the environment.

```
edison05% cime/tools/configure 
ERROR: module command /opt/modules/default/bin/modulecmd python load pe_archive/append-path cray-mpich/7.6.0 PrgEnv-intel/6.0.4 failed with message:
utility.c(2263):ERROR:50: Cannot open file '/usr/common/software/modulefiles/pe_archive/.version' for 'reading'
locate_module.c(1241):ERROR:106: Magic cookie '#%Module' missing in '/usr/common/software/modulefiles/pe_archive/.version'
1.483u 0.469s 0:01.97 98.4% 0pf+0w
```

tangq commented:
@ndkeen , just to add to this issue...my errors on cori-knl today is a bit different:

ERROR: module command /opt/modules/default/bin/modulecmd python swap cray-mpich cray-mpich/7.6.2 failed with message:
cray-mpich(3):ERROR:105: Unable to locate a modulefile for 'cray-mpich/7.6.2'

ndkeen commented:
Helen (at NERSC) replied. I can verify that the simple module commands are now working for me.

```
Click here to view: INC0136923
Comments:
2019-04-11 09:25:24 PDT - Yun (Helen) He Additional comments
Hi Noel,
 
Sorry I changed the default version to be append-path for both Edison and Cori yesterday.
 
On Edison, there was no default version previously. I added the ".version" file, but missed setting the permisison to be global readable.
 
It is fixed now. Please try again.
 
Helen
```

tangq commented:
Hi @ndkeen , I tried again on cori and got the same errors. Is there a way to by pass this issue? Thanks!

ndkeen commented:
OK, I think @tangq has a different error. That version of module cray-mpich was in fact removed from Cori during the SW upgrade yesterday. We would have to use the same pe_archive trick to keep the same module. Or, perhaps even faster test, is to use a version of cray-mpich module that is avail on Cori now. I can help with this.

tangq commented:
Thank you, @ndkeen . Your quick fix passes the configure step. I am testing if it's BFB now. I will update here when the test is done.

ndkeen commented:
Great. Note that I'm still trying to test master after the SW upgrade.

https://acme-climate.atlassian.net/wiki/spaces/EIDMG/pages/959971329/Documenting+potential+changes+to+module+versions+on+Cori+after+SW+maintenance+April+10
